plugins/module_utils/network/sonic/config/sFlow/sFlow.py

```diff
@@ -0,0 +1,376 @@
+            commands.update({"polling_interval":have["polling_interval"]})
+            requests.append({"path":"data/openconfig-sampling-sflow:sampling/sflow/config/polling-interval", "method":"DELETE"})
+
+        if "agent" in want and "agent" in have and want["agent"] == have["agent"]:
+            commands.update({"agent":have["agent"]})
+            requests.append({"path":"data/openconfig-sampling-sflow:sampling/sflow/config/agent", "method":"DELETE"})
+        
+        if ("collectors" in want or len(want) == 0) and "collectors" in have:
+            # here has to be either clear everything or want to clear certain collectors here. not both
+            to_delete_list = have["collectors"]
+            if len(want["collectors"]) > 0:
+                to_delete_list = want["collectors"]
+
+            deleted_list=[]
+            
+            for collector in to_delete_list:
+                found_match = self.contains_collector(have["collectors"], collector)
+                if found_match:
+                    deleted_list.append(collector)
+                    requests.append({"path":"data/openconfig-sampling-sflow:sampling/sflow/collectors/collector="+collector["address"]+","+str(collector["port"])+","+collector["network_instance"], "method":"DELETE"})
+            if len(deleted_list) > 0: 
+                commands.update({"collectors":deleted_list})
+
+        if "interfaces" in want and "interfaces" in have:
+            to_delete_list = have["interfaces"]
+            if len(want["interfaces"]) > 0:
+                to_delete_list = want["interfaces"]
+
+            deleted_list = []
+            for interface in to_delete_list:
+                if "name" not in interface:
+                    continue
+
+                found_interface = self.contains_interface(have["interfaces"], interface)
+
+                if found_interface:
+                    deleted_list.append(interface)
+                    requests.append({"path":"data/openconfig-sampling-sflow:sampling/sflow/interfaces/interface="+interface["name"], "method":"DELETE"})    
+            if len(deleted_list) > 0:
+                commands.update({"interfaces":deleted_list})
+
+        if commands and len(requests) > 0:
+            commands = update_states(commands, "deleted")
+        else:
+            commands = []
+        return commands, requests
+
+    def contains_interface(self, list, search_interface):
+        for interface in list:
+            if interface["name"] == search_interface["name"]:
+                return True
+        return False
+
+    def contains_collector(self, list, search_collector):
+        for collector in list:
+            if collector["address"] == search_collector["address"] and collector["network_instance"] == search_collector["network_instance"]\
+                and collector["port"] == search_collector["port"]:
+                return True
+        return False
+
+    def validate_sflow_args(self, config):
```

Comment:
Please make the capitalization of "sflow" consistent. There are different places where it's called `Sflow` (L30), `sFlow` (L303), or `sflow` (here). It can lead to inadvertent bugs and/or be confusing to maintain going forward if it's inconsistent.

Reply:
fixed some of the inconsistencies in method names in next commit.